### PR TITLE
[DifficultyHash] Prepare for difficulty multiplier usage

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

This PR prepares for claimed amount calculation to use the `RelayDifficultyMultiplier`.
* It removes the `RelayDifficultyTargetHash` `proof` param.
* Replace `prooftypes.DefaultRelayDifficultyTargetHash` occurrences with `protocol.BaseRelayDifficultyHashBz`.
* Moves the `RleayMiningDifficulty` logic to the `service` module due to cyclic dependencies.
* Rename `num_compute_units` to `num_claimed_compute_units` and add `num_estimated_compute_units` and `claimed_amount_upokt`
* Remove the no longer used tokenomics querier to retrieve a service difficulty target hash.

The PR is only renaming and moving of files most (~13200LOC) changes are auto-generated code.

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [x] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

**Review by bryanchriswhite — COMMENTED** [review]
image

**Review by Olshansk — CHANGES_REQUESTED** [review]
Will TAL after merged with main

**Review by bryanchriswhite — COMMENTED** [review]
image

File: x/proof/types/message_update_param_test.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func TestMsgUpdateParam_ValidateBasic(t *testing.T) {
 			name: "invalid: param name incorrect (non-existent)",
 			msg: MsgUpdateParam{
 				Authority: sample.AccAddress(),
-				Name:      "WRONG_relay_difficulty_target_hash",
+				Name:      "WRONG_proof_missing_penalty",
```
Comment (bryanchriswhite) [inline x/proof/types/message_update_param_test.go:31]: For consistency:

```suggestion
				Name:      "non_existent",
```

See [the updated params guide](https://github.com/pokt-network/poktroll/blob/5c995f12ea6957d57c7350f9186ff15bddcc971c/docusaurus/docs/develop/developer_guide/adding_params.md#52-parameter-update-tests):

image

File: x/service/types/genesis_test.go [hunk 2]
```diff
@@ -45,10 +45,33 @@ func TestGenesisState_Validate(t *testing.T) {
...
+					{
+						ServiceId: "1",
+					},
+				},
 				// this line is used by starport scaffolding # types/genesis/validField
 			},
 			expectedErr: nil,
 		},
+		{
+			desc: "invalid - duplicated relayMiningDifficulty",
```
Comment (bryanchriswhite) [inline x/service/types/genesis_test.go:61]: :ok_hand:

File: x/service/types/genesis.go [hunk 3]
```diff
@@ -29,6 +30,17 @@ func (gs GenesisState) Validate() error {
...
+	// Check for duplicated index in relayMiningDifficulty
+	relayMiningDifficultyIndexMap := make(map[string]struct{})
+
+	for _, elem := range gs.RelayMiningDifficultyList {
+		index := string(RelayMiningDifficultyKey(elem.ServiceId))
+		if _, ok := relayMiningDifficultyIndexMap[index]; ok {
+			return ErrServiceDuplicateIndex.Wrapf("duplicated index for relayMiningDifficulty: %s", index)
+		}
+		relayMiningDifficultyIndexMap[index] = struct{}{}
+	}
```
Comment (bryanchriswhite) [inline x/service/types/genesis.go:43]: Wdyt about extracting these to methods (e.g. `#validateServiceList()` & `#validateRelayMiningDifficultyList()`)?

**Review by bryanchriswhite — APPROVED** [review]
Preemptively approving! :raised_hands: 

Just a couple questions and small suggestions - otherwise this LGTM! :ship:

File: testutil/integration/app.go [hunk 4]
```diff
@@ -317,6 +317,7 @@ func NewCompleteIntegrationApp(t *testing.T, opts ...IntegrationAppOptionFn) *Ap
 		apptypes.ModuleName:        {authtypes.Minter, authtypes.Burner, authtypes.Staking},
 		suppliertypes.ModuleName:   {authtypes.Minter, authtypes.Burner, authtypes.Staking},
 		prooftypes.ModuleName:      {authtypes.Minter, authtypes.Burner},
+		servicetypes.ModuleName:    {authtypes.Minter, authtypes.Burner},
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply testutil/integration/app.go:320]: Good catch! It was unnecessary.

**Pull request merged**